src/Workspaces/Remote/Core/Diagnostics/DiagnosticComputer.cs

```diff
@@ -169,8 +169,11 @@ private ImmutableArray<DiagnosticAnalyzer> GetAnalyzers(BidirectionalMap<string,
 
         private OptionSet MergeOptions(OptionSet workspaceOptions, OptionSet userOptions)
         {
+            if (!(userOptions is IInternalOptionSet internalUserOptions))
```

Comment:
Why not make the parameter type `IInternalOptionSet` here?

Reply:
My guess is because the parameter through all these calls is an `OptionSet`, and fixing all the callers is either a breaking change or just a huge pain.

Reply:
Here is a perfect example. This method is private hence no IVT issues. Why not just take `IInternalOptionSet` as the parameter type? 

---
In reply to: [191942919](https://github.com/dotnet/roslyn/pull/27263#discussion_r191942919) [](ancestors = 191942919)